app/models/post_action_type.rb

```diff
@@ -33,54 +49,79 @@ def types
     def expire_cache
       Discourse.redis.keys("post_action_types_*").each { |key| Discourse.redis.del(key) }
       Discourse.redis.keys("post_action_flag_types_*").each { |key| Discourse.redis.del(key) }
+      Discourse.redis.del(POST_ACTION_TYPE_ALL_FLAGS_KEY)
+      Discourse.redis.del(POST_ACTION_TYPE_PUBLIC_TYPE_IDS_KEY)
     end
 
     def reload_types
       @flag_settings = FlagSettings.new
-      ReviewableScore.reload_types
       PostActionType.new.expire_cache
+      PostActionType.expire_cache
+      ReviewableScore.reload_types
     end
 
     def overridden_by_plugin_or_skipped_db?
       flag_settings.flag_types.present? || GlobalSetting.skip_db?
     end
 
     def all_flags
-      Flag.unscoped.order(:position).all
+      cached_all_flags = Discourse.redis.get(PostActionType::POST_ACTION_TYPE_ALL_FLAGS_KEY)
```

Comment:
@lis2 I think we can use the [Cache class](https://github.com/discourse/discourse/blob/main/lib/cache.rb) instead of working with Redis directly. This should simply the code here.